### PR TITLE
feat: Add entry point file

### PR DESCRIPTION
**Pull request by adrien2p**

### CODE REVIEW

**Comment by adrien2p** [comment]
@patrick-medusajs @olivermrbl if you could have a look

File: index.js
Comment (adrien2p) [inline index.js]: self-review: looks like there is some indent issue on my display, someone see the same?

Comment (olivermrbl) [reply index.js]: I see the same, but I think it's a GitHub UI bug.

File: package.json
Comment (olivermrbl) [inline package.json:24]: thought: I am pretty sure `cross-env` is not needed for these scripts, though I guess it's fine to keep it in as a safeguard.

Comment (adrien2p) [reply package.json:24]: yeah I ve put it everywhere as a safe guard if someone want to change the scripts

**Review by olivermrbl — APPROVED** [review]
LGTM - pending @patrick-medusajs's approval

File: index.js
Comment (patrick-medusajs) [reply index.js]: We have a mix of tab (4) and space (2) indentation.
Screenshot 2023-04-13 at 9 01 48 AM

File: index.js
Comment (patrick-medusajs) [inline index.js:4]: thought: In my experiment, I decided to simply surface the `medusa start` command. The reasoning was around maintainability since both `medusa start` (official way from our docs) and `index.js` would follow the same code path. That being said, I didn't explore new opportunities from a different approach. Againg, just a thought.
```javascript
const start = require("@medusajs/medusa/dist/commands/start").default;
const run = async () => {
  const port = process.env.PORT || 9000
  const directory = process.cwd();
  await start({ port, directory });
}
void (async () => {
  await run()
})()
```

**Review by patrick-medusajs — COMMENTED** [review]
Left something to consider and I figured out the indentation issue.

File: index.js
Comment (adrien2p) [reply index.js:4]: The idea of the entry point is to also have the ability to access express directly, which is not the case by using the start command here, though, i am fine with both approaches but I thought that mine would explicit what is today implicit and allow easier customisation from the end user as it is often the needs as we have seen with lots of companies. Wdyt?

File: index.js
Comment (adrien2p) [reply index.js]: Oh 😦 how did that happened, i ll fix it 💪

File: index.js
Comment (patrick-medusajs) [reply index.js:4]: That's what I assumed but I had to ask. Let's keep your version.

**Review by patrick-medusajs — APPROVED** [review]
LGTM

File: index.js
Comment (adrien2p) [reply index.js]: should be fixed now :)

**Comment by adrien2p** [comment]
@olivermrbl should I merge?

**Pull request merged**
